test/test_atomic.cu

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2020 CERN
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * @file test_atomic.cu
+ * @brief Unit test for atomic operations.
+ * @author Andrei Gheata (andrei.gheata@cern.ch)
+ */
+
+#include <iostream>
+#include <cassert>
+#include <AdePT/Atomic.h>
+#include <VecGeom/backend/cuda/Interface.h>
+
+// Example data structure containing several atomics
+struct SomeStruct {
+  adept::Atomic_t<int> var_int{nullptr};
+  adept::Atomic_t<float> var_float{nullptr};
+
+  void EmplaceDataAt(void *addr)
+  {
+    char *add_next = (char *)addr;
```

Comment:
Why `char*` and not `void*` (or `(u)intptr_t` for that matter)?

Reply:
This is outdated, the method was removed